pkg/loop/internal/broker.go

```diff
@@ -34,28 +34,39 @@ func (a *atomicBroker) Accept(id uint32) (net.Listener, error) {
 	return a.load().Accept(id)
 }
 
-func (a *atomicBroker) Dial(id uint32) (conn *grpc.ClientConn, err error) {
-	return a.load().Dial(id)
+func (a *atomicBroker) DialWithOptions(id uint32, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
+	return a.load().DialWithOptions(id, opts...)
 }
 
 func (a *atomicBroker) NextId() uint32 {
 	return a.load().NextId()
 }
 
+// Telemetry has grpc client and server options.
+type Telemetry struct {
+	DialOpts  []grpc.DialOption
+	NewServer func([]grpc.ServerOption) *grpc.Server // optional
```

Comment:
can you add a comment for the use case of the optional NewServer func? why would i want to provide it?